### PR TITLE
Fix installing private data in buildbot fab definition.

### PR DESCRIPTION
**Pull request by adiroiban**
Problem
======

buildmaster private date for stating/testing is a duplicate of the actual private data and get out of sync.

Changes
=======

Update private data with latest buildslaves which are present in production.

As a drive by I have updated the base service code to create the config folder.

Instead of using installPrivateData and installTestData, I just want for installPrivateData ... when it is False it will install the non-private stuff.

For now we need to manually keep in sync the production and stating private data... but I hope that now that we have the vagrant vm available it is much easier to detect out of sync databases.

In the future we might think of a way to reduce the duplication.... but for now I have no idea how to do it.

How to test
=========

On the vagrant, make sure /srv/bb-master folder is clean.

Run `fab -i ~/.vagrant.d/insecure_private_key config.vagrant buildbot.install`  and then check http://172.16.255.140:8080/buildslaves .. it should start and all slaves should have a builder

### CODE REVIEW

**Comment by tomprince** [comment]
>  For now we need to manually keep in sync the production and stating private data

One way to address this, is to run `buildbot checkconfig`, using the sample data. That should at least catch the case where the sample data is out-of-date.

File: braid/twisted/service.py [hunk 1]
```diff
@@ -54,10 +54,12 @@ def bootstrap(self, python='pypy'):
...
 
             # Create base directory setup
-            run('/bin/mkdir -p {} {} {}'.format(
+            run('/bin/mkdir -p {} {} {} {}'.format(
                 self.runDir,
                 self.logDir,
-                self.binDir))
+                self.binDir,
+                self.configDir,
+                ))
```
Comment (glyph) [inline braid/twisted/service.py]: There are 4 `{}`s here and 5 parameters to `.format`, which seems like a problem.

Comment (adiroiban) [inline braid/twisted/service.py]: I only see 4 {}s and 4 parameters ... I have run this code on my vagrant vm and it works.

Comment (glyph) [inline braid/twisted/service.py]: there are 5 lights!@#!@#!@#

(OK, I have no idea what I was thinking, you're right.)

**Comment by hawkowl** [comment]
I think this might need another update (for the Fedora buildbots), then feel free to land.

**Comment by adiroiban** [comment]
OK. I got this running on my vagrant VM... with ansible and all the stuff :)

**Pull request merged**